src/components/partials/UserContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, useState } from 'react'
+
+type userType = {
+  username: string
+  auth: boolean
+}
+
+export type UserContextType = {
+  user: userType
+  login: (username: string) => void
+  logout: () => void
+}
+
+const UserContext = createContext<UserContextType | null>(null)
+
+const UserProvider = ({ children }: { children: React.ReactElement }) => {
+  // User is the name of the "data" that gets stored in context
+  const [user, setUser] = useState<userType>({ username: 'kitty', auth: false })
+
+  // Login updates the user data with a name parameter
+  const login = (username: string) => {
+    setUser({
+      username: username,
+      auth: true,
+    })
+  }
+
+  // Logout updates the user data to default
+  const logout = () => {
+    setUser({
+      username: '',
+      auth: false,
```

Comment:
We can just setUser(null) and each component just check if user is null or not to verify if logged in

Reply:
I think the reason why i opted to do 'auth' as a new field instead of using existing Model is because when user is null, I get a Type error on other component like Navbar that use "user.username" in navigate links that "user might be null", on UI it is conditionally rendered to display none if user is null but the code itself gives me type error